File: dotnet/DesignPatterns/BehavioralDesignPatterns/VisitorDesignPattern/VisitorDesignPattern.Example1/IVisitor.cs
namespace VisitorDesignPattern.Example1;

/// <summary>
/// Visitor interface that declares visit methods for each type of product element.
/// Each concrete visitor will implement different operations for each product type.
/// </summary>
public interface IVisitor
{
    void Visit(ElectronicsProduct product);
    void Visit(FoodProduct product);
    void Visit(ClothingProduct product);
}

File: nodejs/DesignPatterns/BehavioralDesignPatterns/VisitorDesignPattern/VisitorDesignPattern.Example1/src/IVisitor.ts
import { ElectronicsProduct } from './ElectronicsProduct';
import { FoodProduct } from './FoodProduct';
import { ClothingProduct } from './ClothingProduct';

export interface IVisitor {
  visitElectronics(product: ElectronicsProduct): void;
  visitFood(product: FoodProduct): void;
  visitClothing(product: ClothingProduct): void;
}
